lib/autoupdate/package_url.go

```diff
@@ -52,6 +57,82 @@ const (
 	{{- end }}`
 )
 
+// DirFlag returns the directory path representation of a single InstallFlag flag, or "unknown".
```

Comment:
How much of this code is reusable for client tools auto-updates? Would it makes sense to define a different Revision type here and then convert?

Reply:
Idea: what if we leave `Revision` in `lib/autoupdate/agent`, and have `MakeURL` take version and flags separately? The `InstallFlags` methods seem relatively reusable, but the `Revision` methods seem specific to agent updates.

Unless you plan to use the `Revision` methods for client tools in the future?

Reply:
for client-tools we use just a rev structure and flags, let me try to minimize the changes